### PR TITLE
feat(inputs.prometheus): use system wide proxy settings

### LINKED ISSUE #11728: use system wide proxy settings for prometheus input plugin
### Use Case

The prometheus input plugin does not read the system wide proxy settings. But it should, analogous to other plugins or any other http client. 

Some other plugins additionaly allow to set the proxy explicitly. "proxy = ..." If not set, the system wide proxy settings are used. I don't know if this is really necessary.

### Expected behavior

read system proxy settings

### Actual behavior

pull request follows

### Additional info

_No response_

### PR DESCRIPTION
**Pull request by erwiese**
resolves #11728

### CODE REVIEW

**Review by powersj — CHANGES_REQUESTED** [review]
Hi,

Thanks for taking the time to put up a PR.

In Telegraf, we already have a standard HTTP client config that allows users to set the proxy as a part of their configuration. It would be good to use the same behavior across plugins.

Take a look at the HTTP plugin in how it [creates a client](https://github.com/influxdata/telegraf/blob/master/plugins/inputs/http/http.go#L57) with the additional `httpconfig.HTTPClientConfig`.

As well, we would to [add the configuration options](https://github.com/influxdata/telegraf/blob/master/plugins/inputs/http/sample.conf#L35-L36) for the proxy to the plugin configuration.

**Comment by erwiese** [comment]
Hi, could you already review my changes? or anything else missing?

File: plugins/inputs/prometheus/prometheus.go [hunk 1]
```diff
@@ -216,16 +217,15 @@ func (p *Prometheus) GetAllURLs() (map[string]URLAndAddress, error) {
...
+	ctx := context.Background()
+	client, err := p.HTTPClientConfig.CreateClient(ctx, p.Log)
+	if err != nil {
+		return err
+	}
+	p.client = client
+	p.headers = map[string]string{
+		"User-Agent": internal.ProductToken(),
+		"Accept":     acceptHeader,
 	}
```
Comment (srebhan) [inline plugins/inputs/prometheus/prometheus.go:229]: Please move this to an `Init() error` function as it isn't really _required_ to recreate the client in every `Gather()` cycle... The `Init()` function is then called automatically by Telegraf after setting all user-configuration-options.

**Review by srebhan — COMMENTED** [review]
@erwiese, your change looks a lot better this way. Thanks for updating the PR! I have one request left to avoid recreating the `client` over and over. Can you please take a look?

**Comment by Hipska** [comment]
Shouldn't the unix socket client stuff also be added to Init in this case? It is also making a new client on each interval + URL. Both (http & unix) can be placed in the same `p.client`. Furthermore the `response_timeout` can be deprecated in favour of `timeout`.

**Review by powersj — APPROVED** [review]

**Comment by Hipska** [comment]
`response_time` could be marked as deprecated and still being used when `timeout` is not set..

**Review by Hipska — APPROVED** [review]

**Review by MyaLongmire — APPROVED** [review]

**Pull request merged**